src/components/MapView.tsx

```diff
@@ -43,6 +65,85 @@ const MapComponent = ({
 		}
 	};
 
+	/**
+	 * Animates markers from their original locations to their new locations
+	 * upon markers' location change
+	 */
+	useEffect(() => {
+		// Create array to hold any new markers added
+		let newMarkers: MarkerData[] = [];
+
+		// For each marker in the main marker array in props
+		for (const m of markers) {
+			if (m._id in animatedLocations) {
+				// Animate coordinates from current location in AnimatedRegion
+				// object to new location updated in the marker array's object
+				const coordinateConfig = {
+					latitude: m.location.latitude,
+					longitude: m.location.longitude,
+					latitudeDelta: 0,
+					longitudeDelta: 0,
+					useNativeDriver: false,
+					duration: 10000,
+				};
+				// Start the animation
+				animatedLocations[m._id].timing(coordinateConfig).start();
+			} else {
+				// This is a new marker; add it to the new marker array
+				newMarkers.push(m);
+			}
+		}
+		// If a new marker was added to the marker array in props
+		if (newMarkers.length > 0) {
+			// Create a copy of the animatedLocations array from the component
+			// state, add all new markers' AnimatedRegion object to this copy,
+			// and set animatedLocations state to this copy
+			let animatedLocationsCopy = { ...animatedLocations };
+			for (const m of newMarkers) {
+				const animatedM = new AnimatedRegion({
+					latitude: m.location.latitude,
+					longitude: m.location.longitude,
+					latitudeDelta: 0,
+					longitudeDelta: 0,
+				});
+				animatedLocationsCopy[m._id] = animatedM;
+			}
+			setAnimatedLocations(animatedLocationsCopy);
+		}
+		// Remove AnimatedRegions whose markers have been removed
+		return function cleanup() {
```

Comment:
`useEffect` will run the cleanup when the entire component unmounts, which I don't think is exactly what you want?

Since you set state at the end, which is kind of a noop since the component is set to unmount.
I think if we want to handle removing markers, it should be in the main `useEffect` instead of the cleanup funciton.